### PR TITLE
Ksenia Chugunova - UI Design Patterns Challenge. 

### PR DESCRIPTION
**Pull request by kseniaChugunova**

### CODE REVIEW

File: MVP Module/Source/Feed UI/Views/Feed.storyboard [hunk 1]
```diff
@@ -14,19 +14,41 @@
                         <rect key="frame" x="0.0" y="0.0" width="414" height="896"/>
                         <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
                         <color key="backgroundColor" white="1" alpha="1" colorSpace="custom" customColorSpace="genericGamma22GrayColorSpace"/>
-                        <view key="tableHeaderView" contentMode="scaleToFill" id="clu-9B-M8g">
-                            <rect key="frame" x="0.0" y="0.0" width="414" height="16"/>
+                        <view key="tableHeaderView" contentMode="scaleToFill" id="vKc-bt-vSm" customClass="ErrorView" customModule="MVC" customModuleProvider="target">
```
Comment (thiagopenna) [inline MVP Module/Source/Feed UI/Views/Feed.storyboard:17]: This storyboard is in the MVP solution, so the `customModule` should be set to MVP.

```suggestion
                        <view key="tableHeaderView" contentMode="scaleToFill" id="vKc-bt-vSm" customClass="ErrorView" customModule="MVP" customModuleProvider="target">
```

File: MVVM Module/Source/Feed UI/Views/Feed.storyboard [hunk 2]
```diff
@@ -14,19 +14,41 @@
                         <rect key="frame" x="0.0" y="0.0" width="414" height="896"/>
                         <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
                         <color key="backgroundColor" white="1" alpha="1" colorSpace="custom" customColorSpace="genericGamma22GrayColorSpace"/>
-                        <view key="tableHeaderView" contentMode="scaleToFill" id="clu-9B-M8g">
-                            <rect key="frame" x="0.0" y="0.0" width="414" height="16"/>
+                        <view key="tableHeaderView" contentMode="scaleToFill" id="NVf-pe-klG" customClass="ErrorView" customModule="MVC" customModuleProvider="target">
```
Comment (thiagopenna) [inline MVVM Module/Source/Feed UI/Views/Feed.storyboard:17]: This storyboard is in the MVVM solution, so the `customModule` should be set to MVP.

```suggestion
                        <view key="tableHeaderView" contentMode="scaleToFill" id="NVf-pe-klG" customClass="ErrorView" customModule="MVVM" customModuleProvider="target">
```

File: MVP Module/Source/Feed Presentation/FeedLoadingErrorViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,11 @@
+//
+//  FeedLoadingErrorViewModel.swift
+//  MVP
+//
+//  Created by Ksenia on 21.07.2021.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+struct FeedLoadingErrorViewModel {
+	let error: String
```
Comment (thiagopenna) [inline MVP Module/Source/Feed Presentation/FeedLoadingErrorViewModel.swift:10]: The `error` could be an optional `String?`, to better represent the information that the view can display. Instead of the `View` protocol defining two methods (`show` and `hide`), we can accomplish the same behavior displaying a `ViewModel` with an optional value.

For example, in the `didStartLoadingFeed()` method, the Presenter can call  `feedErrorView.display(FeedErrorViewModel(message: .none))` to hide the error.
```suggestion
	let error: String?
```

File: MVP Module/Source/Feed Presentation/FeedLoadingErrorViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,11 @@
+//
+//  FeedLoadingErrorViewModel.swift
+//  MVP
+//
+//  Created by Ksenia on 21.07.2021.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+struct FeedLoadingErrorViewModel {
```
Comment (thiagopenna) [inline MVP Module/Source/Feed Presentation/FeedLoadingErrorViewModel.swift:9]: The concept of `loading` and `error` should be separate since `errors` are not just for loading errors. We can represent errors for many other operations. I suggest renaming this ViewModel:
```suggestion
struct FeedErrorViewModel {
```

**Review by thiagopenna — COMMENTED** [review]
Hi @kseniaChugunova,

This is Penna, I am an instructor at Essential Developer.

Congratulations, you've passed the challenge 🥇

- All tests are passing ✅
- Your code is clean and handles edge cases ✅
- The UI and animations look and behave as expected ✅
- Light and Dark mode support ✅
- Added an extra language ✅

I left some review comments. Feel free to push changes.

Great job!
